### PR TITLE
signed listeners to send email for completion (both pass and fail)

### PR DESCRIPTION
**Pull request by anrao19**
Signed-off-by: Anuchaithra Rao <anrao@redhat.com>

# Description
Problem: listener job failed on 5.3 RC, But no notification sent out on release critical stage(RC)

Solution/Ideas/Improvements: Enable email notification from the jenkins job even if we build failure.

Failure: send out alert message in the email
Success: Send out success email
Common atributes should be included: build information.

PASS: https://jenkins.ceph.redhat.com/job/rhceph-signed-compose-listener/52/console
Screenshot from 2023-01-20 09-48-46

FAIL: https://jenkins.ceph.redhat.com/job/rhceph-signed-compose-listener/53/console
Screenshot from 2023-01-20 10-32-39

Latest Mail
Pass:
Screenshot from 2023-01-23 21-24-01

Fail:
Screenshot from 2023-01-23 21-24-21

Latest Log:
Pass: 
Screenshot from 2023-01-30 12-15-06

Fail: 
Screenshot from 2023-01-30 12-14-57

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Review by tintumathew10 — COMMENTED** [review]
LGTM

File: pipeline/signed-compose-listener.groovy [hunk 1]
```diff
@@ -115,19 +116,24 @@ node("rhel-8-medium || ceph-qe-ci") {
...
         }
-
-        // notify about failure
-        currentBuild.result = "FAILURE"
-        def failureReason = err.getMessage()
-        def subject =  "[CEPHCI-PIPELINE-ALERT] [JOB-FAILURE] - ${env.JOB_NAME}/${env.BUILD_NUMBER}"
-        def body = "<body><h3><u>Job Failure</u></h3></p>"
-        body += "<dl><dt>Jenkins Build:</dt><dd>${env.BUILD_URL}</dd>"
-        body += "<dt>Failure Reason:</dt><dd>${failureReason}</dd></dl></body>"
+    } finally {
```
Comment (sunilkumarn417) [inline pipeline/signed-compose-listener.groovy:127]: Is there a way to fetch RC build information and put it in tthe email?

File: pipeline/signed-compose-listener.groovy [hunk 2]
```diff
@@ -115,19 +116,24 @@ node("rhel-8-medium || ceph-qe-ci") {
...
+        def subject =  "[CEPHCI-PIPELINE] ${env.JOB_NAME}/${env.BUILD_NUMBER} - ${currentBuild.result}"
+        def body = "<body><h3><u>Job ${currentBuild.result}</u></h3></p>"
```
Comment (sunilkumarn417) [inline pipeline/signed-compose-listener.groovy:129]: I believe `</p>` is not needed.

File: pipeline/signed-compose-listener.groovy (hunk 1 above)
Comment (sunilkumarn417) [reply pipeline/signed-compose-listener.groovy:127]: Just add some more information about RC and it would be good if we mention RC in Subject line.

Comment (anrao19) [reply pipeline/signed-compose-listener.groovy:128]: Made Changes as per comment

File: pipeline/signed-container-image-listener.groovy [hunk 3]
```diff
@@ -118,26 +120,37 @@ node("rhel-8-medium || ceph-qe-ci") {
...
+            println "Failure Reason: ${failureReason}"
         }
+    } finally {
+        def body = "<body><h3><u>Job ${currentBuild.result}</u></h3>"
+        body += "<dl><dt>RHCS Version:</dt><dd>RHCEPH-${versions.major_version}.${versions.minor_version}</dd>"
+        body += "<dt>Ceph Version:</dt><dd>${cephVersion}</dd>"
+        body += "<dt>Compose Label:</dt><dd>${releaseMap.rc.'compose-label'}</dd>"
+        body += "<dt>Container Image:</dt><dd>${containerImage}</dd>"
+        def sub = ""
+        if (currentBuild.result == "FAILURE") {
```
Comment (sunilkumarn417) [inline pipeline/signed-container-image-listener.groovy:138]: I think `ABORTED` should also considered.

File: pipeline/signed-compose-listener.groovy [hunk 4]
```diff
@@ -115,19 +117,28 @@ node("rhel-8-medium || ceph-qe-ci") {
...
-        body += "<dl><dt>Jenkins Build:</dt><dd>${env.BUILD_URL}</dd>"
-        body += "<dt>Failure Reason:</dt><dd>${failureReason}</dd></dl></body>"
+    } finally {
+        def body = "<body><h3><u>Job ${currentBuild.result}</u></h3>"
+        body += "<dl><dt>RHCS Version:</dt><dd>RHCEPH-${versions.major_version}.${versions.minor_version}</dd>"
+        body += "<dt>Ceph Version:</dt><dd>${cephVersion}</dd>"
+        body += "<dt>Compose Label:</dt><dd>${cimsg['compose-label']}</dd>"
+        body += "<dt>Compose URL:</dt><dd>${composeUrl}</dd>"
+        def sub = ""
+        if (currentBuild.result == "FAILURE") {
```
Comment (sunilkumarn417) [inline pipeline/signed-compose-listener.groovy:135]: `ABORTED` should also be considered.

**Review by sunilkumarn417 — COMMENTED** [review]
please retest this once with changes.

File: pipeline/signed-compose-listener.groovy (hunk 4 above)
Comment (anrao19) [reply pipeline/signed-compose-listener.groovy]: Made changes

**Comment by anrao19** [comment]
> please retest this once with changes.

Retested and Uploaded log

**Review by sunilkumarn417 — APPROVED** [review]

**Pull request merged**